recipes/glibmm/all/conanfile.py

```diff
@@ -43,129 +51,115 @@ def _giomm_lib(self):
-
+        self.requires("glib/2.75.2")
         if self._abi_version == "2.68":
             self.requires("libsigcpp/3.0.7")
         else:
             self.requires("libsigcpp/2.10.8")
 
     def source(self):
-        files.get(self, **self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
+        get(self, **self.conan_data["sources"][self.version], strip_root=True, destination=self.source_folder)
+
+    def generate(self):
+        deps = PkgConfigDeps(self)
+        deps.generate()
+
+        tc = MesonToolchain(self)
+        tc.project_options.update({
+            "build-examples": "false",
+            "build-documentation": "false",
+            "msvc14x-parallel-installable": "false",
+            "default_library": "shared" if self.options.shared else "static"
+        })
+        tc.generate()
+
+        env = VirtualBuildEnv(self)
+        env.generate()
 
     def _patch_sources(self):
-        files.apply_conandata_patches(self)
-        meson_build = os.path.join(self._source_subfolder, "meson.build")
-        files.replace_in_file(self, meson_build, "subdir('tests')", "")
-        if microsoft.is_msvc(self):
+        apply_conandata_patches(self)
+        meson_build = os.path.join(self.source_folder, "meson.build")
+        replace_in_file(self, meson_build, "subdir('tests')", "")
+        if is_msvc(self):
             # GLiBMM_GEN_EXTRA_DEFS_STATIC is not defined anywhere and is not
             # used anywhere except here
             # when building a static build !defined(GLiBMM_GEN_EXTRA_DEFS_STATIC)
             # evaluates to 0
             if not self.options.shared:
-                files.replace_in_file(self,
-                    os.path.join(self._source_subfolder, "tools",
-                                 "extra_defs_gen", "generate_extra_defs.h"),
-                    "#if defined (_MSC_VER) && !defined (GLIBMM_GEN_EXTRA_DEFS_STATIC)",
-                    "#if 0",
-                )
+                replace_in_file(self,
+                                      os.path.join(self.source_folder, "tools",
+                                                   "extra_defs_gen", "generate_extra_defs.h"),
+                                      "#if defined (_MSC_VER) && !defined (GLIBMM_GEN_EXTRA_DEFS_STATIC)",
+                                      "#if 0",
+                                      )
 
             # when using cpp_std=c++NM the /permissive- flag is added which
             # attempts enforcing standard conformant c++ code
             # the problem is that older versions of Windows SDK is not standard
             # conformant! see:
             # https://developercommunity.visualstudio.com/t/error-c2760-in-combaseapih-with-windows-sdk-81-and/185399
-            files.replace_in_file(self, meson_build, "cpp_std=c++", "cpp_std=vc++")
+            replace_in_file(self, meson_build, "cpp_std=c++", "cpp_std=vc++")
 
     def configure(self):
```

Comment:
I wouldn't be against reorder a little bit methods by order of execution, it's super hard to follow global logic of this recipe with configur